### DISCUSSION TITLE
Error 'FAILED_TO_DESERIALIZE_ARGUMENT' with Array Parameter Exceeding 10 Values in Aptos SDK v1.17.0

### DISCUSSION THREAD

**Discussion by ajselvan** [aptos-labs/aptos-developer-discussions#503] [Questions] [answered]
### Describe your question in detail.

Hi , we are encountering the error 'FAILED_TO_DESERIALIZE_ARGUMENT' when passing function arguments. Specifically, if an array parameter contains more than 10 values, this error occurs in the SDK during the signAndSubmitTransaction process. Could you please assist us with resolving this issue.
version: @aptos-labs/ts-sdk": "^1.17.0

### What error, if any, are you getting?

FAILED_TO_DESERIALIZE_ARGUMENT

### What have you tried or looked at?  Or how can we reproduce the error?

when passing function arguments. Specifically, if an array parameter contains more than 10 values

### Which operating system are you using?

Windows

### Which SDK or tool are you using? (if any)

TypeScript SDK

### Describe your environment or tooling in detail

SDK --> version: @aptos-labs/ts-sdk": "^1.17.0

**Comment by 0xbrw** [comment]
a few options you can try. 
- Our latest ts-sdk is on v1.30.0: https://aptos-labs.github.io/aptos-ts-sdk/@aptos-labs/ts-sdk-1.30.0/
- You can try running `npm install @aptos-labs/ts-sdk@latest`
- you could also try splitting you array into smaller chunks and submit multiple transactions

cc @0xmaayan

**Comment by 0xmaayan** [comment]
@ajselvan Could you help with providing a code snippet so we can try and reproduce the error

**Reply by ajselvan** [reply]
Hi @0xmaayan this is my code snippet:

```
import { AptosConfig, Aptos, Network, Account, Ed25519PrivateKey } from "@aptos-labs/ts-sdk";
import dotenv from "dotenv";
import { PerpsMarkets } from "../src/econia/markets/perpsMarket";
dotenv.config();

async function main(): Promise<void> {

    // Setup the client
    const config = new AptosConfig({ network: Network.TESTNET });
    const aptos = new Aptos(config);
    const account = Account.fromPrivateKey({
        privateKey: new Ed25519PrivateKey(process.env.APTOS_PRIVATEKEY || ''),
    }); 
    const marketId = 66;
    const orderTypes = [true, true, true, true, true, true, true, true, true, true, true, true]
    const tradeSides = [true, true, true, true, true, true, true, true, true, true, true, true];
    const directions = [true, true, true, true, true, true, true, true, true, true, true, true]; //buy
    const sizes = [1000, 2000, 1000, 2000, 1000, 2000, 1000, 2000, 1000, 2000, 1000, 2000]; // 1 APT
    const prices = [5000, 6000, 5000, 6000, 5000, 6000, 5000, 6000, 5000, 6000, 5000, 6000]; // 5 USDC
    const leverages = [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2];
    const takeProfits = [undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined];
    const stopLosses = [undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined];
    const restrictions = [undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined];
    
    const perpsMarketsInstance = new PerpsMarkets(process.env.APTOS_PRIVATEKEY || '', Network.TESTNET, process.env.NODEREAL_TESTNET_URL);
    const payload = await perpsMarketsInstance.placeMultipleOrders(marketId, orderTypes, tradeSides, directions, sizes, prices, leverages, restrictions, takeProfits, stopLosses);
    const transactionPayload = await aptos.transaction.build.simple({
        sender: account.accountAddress,
        data: payload.data
    });

    const committedTxn = await aptos.transaction.signAndSubmitTransaction({
        transaction: transactionPayload,
        signer: account,
    });
    console.log(`Submitted transaction: ${committedTxn.hash}`);
    const response = await aptos.waitForTransaction({
        transactionHash: committedTxn.hash,
    });
    console.log("response", response.success);

}

main().catch(error => {
    console.error('An error occurred:', error);
});

```

**Comment by ajselvan** [comment]
Hi @0xmaayan Could you please assist us with resolving this issue.

**Reply by 0xmaayan** [reply]
Thanks, could you share the full transaction payload and the contract address you submit the transaction to?

**Reply by ajselvan** [reply]
Hi @0xmaayan 

### This is my payload :

payload:  {
  success: true,
  message: 'payload built succesfully',
  data: {
    function: '0x1ce63c441256263ea2f45dd2d3a1e68e5843107c65c10b79c7a2d5aa2b7a4e3e::perp_market::place_multiple_orders',
    functionArguments: [
      66,      [Array],
      [Array], [Array],
      [Array], [Array],
      [Array], [Array],
      [Array], [Array]
    ],
    typeArguments: [
      '0x197b9c79f40089f150e3c493edb8cf145efec6bd6b0474f3b08ad738549fa1d1::asset::USDC'
    ]
  }
}

### This is my contract_address :

**0x1ce63c441256263ea2f45dd2d3a1e68e5843107c65c10b79c7a2d5aa2b7a4e3e**

### This is my error txn_hash :

An error occurred: Xr [Error]: Transaction 0xc612de01cd2227707815cb35d595982fef7771b3eef1661aa414f62b826eced1 failed with an error: FAILED_TO_DESERIALIZE_ARGUMENT.

**Comment by gregnazario** [comment, marked as answer]
## I'm getting `FAILED_TO_DESERIALIZE_ARGUMENT` when using a vector with more than `10` structs in it (e.g. `Option`, `Object`, etc.)

### What happened?
The limit on the number of nested or unpacked structs (including in a vector) is 10.

This is from a previous security mitigation in the current VM implementation and for the moment, will stay at that amount.

https://github.com/aptos-labs/aptos-core/pull/8568/files#diff-ce29aaa4f72d88fa625f39d7ba651b6cb38014b3d38326538a630b2107384eabR255-R264

### How can I address this?

There are two paths:
* if you're using `vector<Object<T>>` you can use `vector<address>` instead.
* if you're using `vector<Option<T>>` you can use `vector<T>`, and have a sentinel value that isn't used.  For exapmle, if it was `vector<Option<u64>>` you could use a `vector<u64>` and use `0` as a "none" option.
